src/Microsoft.AspNetCore.Razor.Evolution/Legacy/HtmlMarkupParser.cs

```diff
@@ -219,10 +219,13 @@ private void OtherParserBlock()
         {
             AddMarkerSymbolIfNecessary();
             Output(SpanKind.Markup);
+
             using (PushSpanConfig())
             {
                 CodeParser.ParseBlock();
             }
+
+            Span.Start = CurrentLocation;
```

Comment:
We need to initialize the 'start' of the spanbuilder each time we transition between tokenizers. This means we 'catch up' each time the other tokenizer did some work.

Putting this state in a single object would really help, but I didn't want to pull that into this PR.